policy/nomad/source.go

```diff
@@ -102,41 +104,26 @@ func (s *Source) MonitorIDs(ctx context.Context, req policy.MonitorIDsReq) {
 		// Perform a blocking query on the Nomad API that returns a stub list
 		// of scaling policies. The call is done in a goroutine so we can
 		// still listen for the context closing or a reload request.
-		blockingQueryCompleteCh := make(chan struct{})
-		go func() {
-			// Obtain a handler now so we can release the lock before starting
-			// the blocking query.
-			s.nomadLock.RLock()
-			scaling := s.nomad.Scaling()
-			s.nomadLock.RUnlock()
+		//blockingQueryCompleteCh := make(chan struct{})
 
```

Comment:
this comment no longer applies without the goroutine, but why remove the goroutine?

since the API call is a blocking query (right?), which might hold open for a while, we would want to be able to drop it on quit/reload...

Reply:
Before this was a weird logic that the first time the query returned, the `blockingQueryCompleteCh` was closed or a time out and error happend, the intention behind it  seemed to be to ensure queries were possible and the connection was working, so I split the logic, moving the initial call to initialisation, making the  `blockingQueryCompleteCh` unnecessary. The test I removed was checking that logic.